Formula/earthly.rb

```diff
@@ -35,13 +33,9 @@ def install
   end
 
   test do
-    (testpath/"build.earthly").write <<~EOS
-
-      default:
-      \tRUN echo Homebrew
-    EOS
-
-    output = shell_output("#{bin}/earthly --buildkit-host 127.0.0.1 +default 2>&1", 6).strip
-    assert_match "buildkitd failed to start", output
+    # earthly requires docker to run; therefore doing a complete integration test here
+    # is not possible, so instead we simply test the --version command runs
+    output = shell_output("#{bin}/earthly --version").strip
+    assert output.start_with?("earthly version")
```

Comment:
What was wrong with the old test? In my experience it's more likely to catch issues than a version check.

Reply:
The old test was checking that earthly failed to start up (since docker is not available); typically earthly **doesn't** exit with a failure. To complicate matters, the error message it prints out on failure has changed, and is more likely to change in the future than the version command.

> In my experience it's more likely to catch issues than a version check.

If docker were available, then I would completely agree with you. But in this case, neigther the old test (or this version test) is likely to catch many issues.

Reply:
If it's checking for the docker command you can add `depends_on "docker" => :test`. But seeing how we explicitly tell people not to do version tests in our documentation, this change is not okay for me. We'll have to test a command that people are actually likely to use.

Reply:
> If it's checking for the docker command you can add depends_on "docker" => :test

Does this allow running the docker service in addition, e.g. would this allow earthly to run a command such as `docker run --rm alpine echo hello world`?

Reply:
Not on macOS since running docker on macOS requires software that is not in homebrew-core.

Reply:
> Not on macOS since running docker on macOS requires software that is not in homebrew-core.

Thanks for confirming that.

> We'll have to test a command that people are actually likely to use.

All the commands that people actually use (aside from `--version` and `--help`), require interaction with buildkit which earthly starts up via docker. Without the buildkit docker container running, earthly will fail -- I don't see how checking for this failure produces any extra benefit compared to a version check?

Reply:
Because help or version commands generally share no code with the commands that users are interested in so they generally don't throw errors for missing dependencies etc.

Reply:
`earthly` is written in go, all libraries are included via static-compilation, it is not susceptible to such issues.

Can we make a special exception for this case?

Reply:
I'm not sure why you insist on getting an exception if there was a perfectly fine test in the formula already. The rules are there because of Homebrew's experience with package management and unless there is no other way I'd prefer to follow them.

Reply:
> I'm not sure why you insist on getting an exception if there was a perfectly fine test in the formula already

So to double check: you want me to revert to the test that depends on an error occurring within earthly?

Reply:
@SMillerDev I discovered that I can do a `earthly ls` without requiring docker. Will this work instead?

Why this new test compared to the old one:
 - the old test is confusing as it relies on an error occuring
 - the error message has since changed (and would require the test be updated)
 - retry logic in earthly has been introduced, and it now takes 1 minute before the error message is displayed

Fortunately, `earthly ls` works without the above issues, and also executes code in earthly's parser.